README.md
Survive!
==========================================================  
~                                                        ~  
~	You find yourself suddenly thrown into		 ~  
~	a strange cubic world, and you must rely         ~  
~	on your wits and what resources you can          ~  
~	find and harvest. Eating and drinking are        ~  
~	important, without food or water you will        ~  
~	find yourself losing health and won't last       ~  
~	very long. Be Careful, not all foods are         ~  
~	safe to eat, some can cause sickness or          ~  
~	even death. Make yourself a weapon, find         ~  
~	shelter and prepare to survive.                  ~  
~                                                        ~  
==========================================================  

Please Note, this game is heavily under development and has many bugs and unfinished parts of code. Play at your own risk. Also there could be some game breaking changes in the future, though I'll try and use ABMs to change things so if you keep up to date and update regularly problems should take care of themselves.


changelog.txt
2017-01-28:
Many changes to armor.
Added NSSPF by NPX team.
Added the ability to fill a wineskin from the well, and drink machines.

2017-01-13:
Added orienteering mod.
updated mobs.
updated 3d_armor
updated unified_inventory.

2016-10-23:
Updated a few mods.
juice press now can automatically fill a barrel that is below it.

2016-10-04:
Some small tweaks to the drink machines.

2016-09-25:
Juicing machine and juice storage are complete.

2016-09-11:
Added various parts of juicing machines and juices.
updated several mods that had depreciated code.

2016-08-17:
Added Fachwork mod.

2016-08-10:
Tweaked food values more, should be more fair now.

2016-07-30:
Tons of changes, updated several mods.
Added Farming Plus.
Made some updates to the food systems. New textures.
Moved animals into a new Animals mod, with all customized drops.
Updated doors, so now there are the gates from default.

2016-07-27:
Added variations of poultry.
Added some missing textures.
Fixed issue with traps not working properly.
Increased hunger rate, so food is more important.
Changed the berry bushes to grow fruit more often in the summer.

2016-06-29:
Fixed issue with 3d_armor and not having leather armor.

2016-06-22:
Updated Mobs API, Goblins, and 3D_armor

2016-02-15:
Updated the HUD/Hunger mod.

2016-02-11:
Updated mobs again.
Fixed plantlife modpack so the berry bushes actually spawn in world, I think it was a humidity problem.
Removed the pathogen mod, as I heard it caused some crashes and it currently wasn't doing anything.

2016-02-03:
Updated mobs to latest version from TenPlus1

2016-01-17:
Fixed a bug with Unified Inventory that could crash the game.
Worked on adding the textures for the meats.
Discovered that my groups aren't working, can't figure out why either.

2016-01-13:
Tweaked hunger and thirst speed.
Added Goat mob.
Changed around drops for animals.
Added some new foods to food mod.
A few tweaks to plants modpack so berry bushes should start appearing in game again.

2015-12-10:
Updated traps to work correctly for freeing and butchering animals.
Tweaked some grouping for the traps so more baits will bring in game.

2015-12-02:
Added the Bees mod, fixed the leaves to be in the kindling group again.

2015-10-7:
Added in a few mobs, have yet to tweak the spawn rates to make them much rarer. Also want to make them run away from the player, though I don't seem to notice an option for that.

2015-09-22:
Added the ambience mod, to create more immersive gameplay.

2015-09-10:
Major updates on traps. Currently you can only use raw meat to bait them. all buttons should be working in the formspecs.

2015-09-09:
Started work on ABMs for traps and did some minor coding changes in the nodes and formspecs

2015-09-07:
Updated Goblins Mod again.
Added Pathv6 as a new mod.

2015-09-04:
Updated Goblins Mod again.
Snare now has several states with different formspecs for each.

2015-09-02:
Updated More_fire to fix a few missing textures.
Tweaked around grass growth and spread ABMs. Grass will now just grow on soils that have full grass texture on top, which should help with the disparity in looks.
Tweaked the ABM for the trail erosion, dirt nodes only change back into grass if they have air on top of them. This fixes the issue of default:dirt_with_grass showing up under water.
Added model and start of code for a simple snare.
Started working on food poisoning, currently there is some problem with the pathogen actually kicking in, but other coding is all set to go.

2015-08-27:
Updated More_fire and Survival
Added a couple ABMs for grass spreading and growth.

2015-08-26:
Updated Valley_Mapgen.
Added intersecting mod from paramat. Should be easier to find Goblins in underground caves with this.
Updated Default.

2015-08-12:
Well bottoms are now placable on any nodes that belong to the soil group, code also got a huge cleanup from that change.
Smokers are still very hacky, use at your own risk, they don't smoke anything yet, so there really is no point to use them.

2015-08-10:
Put code on github.

2015-08-01:
Added inventory image for smoker.
Created active smoker node, started working on coding.

2015-07-31:
Fixed bug in trail code, it now functions properly for grasses and soils.
Added mesh and texture for the smoker, no coding yet.


mods/mobs/goat.lua

-- Goat by DonBatman/Nathan.S

mobs:register_mob("mobs:goat", {
	-- animal, monster, npc, barbarian
	type = "animal",
	-- aggressive, does 4 damage to player when threatened
	passive = false,
	attack_type = "dogfight",
	damage = 2,
	-- health & armor
	hp_min = 5, hp_max = 15, armor = 200,
	-- textures and model
	collisionbox = {-0.3, -0.01, -0.3, 0.3, 0.75, 0.3},
	visual = "mesh",
	mesh = "mobs_goat.b3d",
	drawtype = "front",
	textures = {
		{"mobs_goat_white.png"},
		{"mobs_goat_brown.png"},
		{"mobs_goat_grey.png"},
	},
	blood_texture = "mobs_blood.png",
	visual_size = {x=2,y=2},
	-- sounds
	makes_footstep_sound = true,
	sounds = {
		random = "mobs_sheep",
	},
	-- speed and jump
	walk_velocity = 1,
	run_velocity = 2,
	jump = true,
	stepheight = 1.1,
	-- drops raw meat when dead
	drops = {
		{name = "mobs:meat_raw",
		chance = 1, min = 2, max = 4},
		{name = "mobs:leather",
		chance =1, min = 1, max = 3},
	},
	-- damaged by
	water_damage = 1,
	lava_damage = 5,
	light_damage = 0,
	-- model animation
	animation = {
		speed_normal = 15,		speed_run = 15,
		stand_start = 1,		stand_end = 60, -- head down/up
		walk_start = 122,		walk_end = 182, -- walk
		run_start = 122,		run_end = 182, -- walk
		punch_start = 246,		punch_end = 330, -- attack
		-- 1-30 head down, 31-60 head up, 61-121 tail wiggle, 185-215 lay down, 216-245 getting up, 331-390 chewing, 391-511 look around
	},
	-- follows wheat
	follow = "farming:wheat", view_range = 7,
	-- replace grass/wheat with air (eat)
	replace_rate = 50,
	replace_what = {"group:flora"},
	replace_with = "air",
})
-- spawn on default;green;prairie grass between 0 and 20 light, 1 in 20000 chance, 1 goat in area up to 31000 in height
mobs:register_spawn("mobs:goat", {"default:dirt_with_grass", "ethereal:green_dirt_top"}, 20, 0, 20000, 1, 31000)
-- register spawn egg
mobs:register_egg("mobs:goat", "Goat", "default_grass.png", 1)
